lib/spack/spack/test/stage.py

```diff
@@ -879,34 +824,19 @@ def test_diystage_preserve_file(self, tmpdir):
             _file.read() == _readme_contents
 
 
-@pytest.fixture
-def tmp_build_stage_nondir(tmpdir):
-    """Establish the temporary build_stage pointing to non-directory."""
-    test_stage_path = tmpdir.join('stage', 'afile')
-    test_stage_path.ensure(dir=False)
-
-    # Set test_stage_path as the default directory to use for test stages.
-    current = spack.config.get('config:build_stage')
-    stage_dir = os.path.dirname(str(test_stage_path))
-    spack.config.set('config', {'build_stage': [stage_dir]}, scope='user')
-
-    yield test_stage_path
-
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
 def test_stage_create_replace_path(tmp_build_stage_dir):
     """Ensure stage creation replaces a non-directory path."""
     _, test_stage_path = tmp_build_stage_dir
```

Comment:
This instance of `tmp_build_stage_dir` can also be removed and replaced with `test_stage_path = spack.stage.get_stage_root()`, although you have to call `stage.destroy` at the end of the test (and I recommend putting that in a `try/finally` block to avoid impacts to other tests.

Reply:
Also relevant to addition of this issue in #13065